### PR TITLE
JBTM-3755 Support for passing participant data to LRA callbacks

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3755

I removed the axes to test because the last commit was a formatting change

I'll follow up with a docs update once it's passed review.

### CODE REVIEW

**Comment by marcosgopen** [comment]
Thanks for your PR Michael! Is this PR related to the topic discussed publicly on zulip (https://narayana.zulipchat.com/#narrow/stream/323714-users/topic/.40Complete.20behaviour.20with.20.40LRA ) ?

File: rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java [hunk 1]
```diff
@@ -759,6 +778,13 @@ private URI enlistCompensator(URI uri, Long timelimit, String linkHeader, String
             }
 
             String recoveryUrl = null;
+                String prevParticipantData = response.getHeaderString(NARAYANA_LRA_PARTICIPANT_DATA_HEADER_NAME);
+
+                if (compensatorData != null && prevParticipantData != null) {
+                    compensatorData.setLength(0);
+                    compensatorData.append(prevParticipantData);
+                }
+
```
Comment (marcosgopen) [inline rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java:787]: What about formatting these lines (removing additional spaces). Thanks

File: rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/resource/ParticipantDataResource.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    public static final String SIMPLE_PARTICIPANT_RESOURCE_PATH = "simple-lra-participant";
+    public static final String START_LRA_PATH = "start-end-lra";
+    public static final String BEGIN_LRA_PATH = "begin-lra";
+    public static final String END_LRA_PATH = "end-lra";
+
+    public static final String CALLS_PATH = "calls";
+
+    private static final String CONTEXT_DATA = "context for doInLRA";
+    private static final String BEGIN_DATA = "context for begin";
+    private static final String END_DATA = "context for end";
```
Comment (marcosgopen) [inline rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/resource/ParticipantDataResource.java:48]: What is the purpose for this END_DATA variable?

**Review by marcosgopen — APPROVED** [review]
Thanks @mmusgrov for this PR. I appreciated the use of the StringBuilder, the method overloading and the orElse method!

**Comment by mmusgrov** [comment]
> Thanks for your PR Michael! Is this PR related to the topic discussed publicly on zulip (https://narayana.zulipchat.com/#narrow/stream/323714-users/topic/.40Complete.20behaviour.20with.20.40LRA ) ?

It is, it took a bit longer than planned.

**Review by jmfinelli — APPROVED** [review]

**Pull request merged**